pyterrier_rag/backend/_base.py

```diff
@@ -63,13 +63,15 @@ def generate(
         *,
         return_logprobs: bool = False,
         max_new_tokens: Optional[int] = None,
+        stop_tokens : Optional[List[str]] = None,
     ) -> List[BackendOutput]:
         """ Generate text from input prompts.
 
         Parameters:
             inps (Union[List[str], List[List[dict]]]): Input prompts as strings or dictionaries. When strings, represent the prompts directly. When a list of dictionaries, represents a sequence of messages (if ``backend.supports_message_input==True``).
             return_logprobs (bool): Whether to return logprobs of generated tokens along with text. (Only available if ``backend.supports_logprobs==True``.)
             max_new_tokens (Optional[int]): Override for max tokens to generate.
+            stop_tokens : Optional[List[str]]: List of tokens at which to stop generation. If None, generation is unconstrained.
```

Comment:
I think this should also specify whether the stop token will be present in the output (I think in the current implementations it's not, right?)

Unconstrained could be confusing, since it's still constrained by the max tokens.